### PR TITLE
Feature: `ArtifactCard` component

### PR DESCRIPTION
**Pull request by znicholasbrown**
Adds a new `ArtifactCard` component used for both lists and summary displays on artifact (results) pages

### CODE REVIEW

File: src/utilities/dates.ts [hunk 1]
```diff
@@ -6,7 +6,7 @@ import { dateFunctions, toDate, formatDateInTimezone } from '@/utilities/timezon
 // https://www.notion.so/prefect/Formatting-Basics-1d4d83bcb2ba471ead90910aeb5913b2?pvs=4#b79673ff9e9b47f58236c449e5fe764a
 const dateFormat = 'MMM do, yyyy'
 const timeFormat = 'hh:mm a'
-const dateTimeFormat = `${dateFormat} ''at'' ${timeFormat}`
+const dateTimeFormat = `${dateFormat} 'at' ${timeFormat}`
```
Comment (znicholasbrown) [inline src/utilities/dates.ts:9]: @pleek91 I think this format wasn't quite correct before but wanted to call it out just in case

Comment (pleek91) [reply src/utilities/dates.ts:9]: I'm not and this wasn't used anywhere when I added it. I misunderstood the date-fns docs example. Thanks for fixing.

File: src/components/ArtifactCard.vue [hunk 2]
```diff
@@ -0,0 +1,232 @@
+<template>
+  <p-card class="artifact-card" :class="classes.root">
+    <div class="artifact-card__body" :class="classes.body">
+      <header class="artifact-card__header" :class="classes.header">
+        <h6 class="artifact-card__subheader" :class="classes.subheader">
+          {{ artifact.type }}
+        </h6>
+        <h3 v-if="hasKey">
+          {{ artifact.key }}
+        </h3>
```
Comment (pleek91) [inline src/components/ArtifactCard.vue:10]: I don't know what this should be but an `h6` followed by an `h3` seems a little off to me? I thought generally headings were supposed to be in order semantically. Could be way off and this is fine.

File: src/components/ArtifactCard.vue [hunk 3]
```diff
@@ -0,0 +1,232 @@
...
+
+        <template v-if="hasRun && crumbs.length">
+          <p-bread-crumbs :crumbs="crumbs" class="artifact-card__bread-crumbs" :class="classes.breadCrumbs" />
+        </template>
+      </header>
+
+      <div class="artifact-card__summary-container" :class="classes.summaryContainer">
+        <div
+          class="artifact-card__summary-item"
+          :title="artifact.updated.toLocaleString()"
```
Comment (pleek91) [inline src/components/ArtifactCard.vue:20]: What's the thought behind these titles attrs? The same info is present on the screen. Won't a screen reader end up reading this date twice?

File: src/components/ArtifactCard.vue [hunk 4]
```diff
@@ -0,0 +1,232 @@
...
+  const hasKey = computed(() => !!props.artifact.key)
+
+  const crumbs = computed<BreadCrumbs>(() => {
+    const internalCrumbs: BreadCrumbs = []
+
+    if (flowRunId.value && !routeIncludes.value.flowRun) {
+      internalCrumbs.push({
+        text: flowRun.value?.name ?? '',
+        to: routes.flowRun(flowRunId.value),
+      })
```
Comment (pleek91) [inline src/components/ArtifactCard.vue:84]: Switching this around a bit will get rid of some checks. (untested, but pretty sure this is correct)
```suggestion
    if (flowRun.value && !routeIncludes.value.flowRun) {
      internalCrumbs.push({
        text: flowRun.value.name,
        to: routes.flowRun(flowRun.value.id),
      })
```

File: src/components/ArtifactCard.vue [hunk 5]
```diff
@@ -0,0 +1,232 @@
...
+    }
+
+    if (taskRunId.value && !routeIncludes.value.taskRun) {
+      internalCrumbs.push({
+        text: taskRun.value?.name ?? '',
+        to: routes.taskRun(taskRunId.value),
+      })
+    }
```
Comment (pleek91) [inline src/components/ArtifactCard.vue:92]: same here

File: src/components/ArtifactCard.vue [hunk 6]
```diff
@@ -0,0 +1,232 @@
...
+  const flowRunId = computed(() => props.artifact.flowRunId)
+  const taskRunId = computed(() => props.artifact.taskRunId)
+
+  const flowRun = useFlowRun(flowRunId)
+  const taskRun = useTaskRun(taskRunId)
+
+  const hasRun = computed(() => !!props.artifact.flowRunId || !!props.artifact.taskRunId)
+  const hasKey = computed(() => !!props.artifact.key)
+
+  const crumbs = computed<BreadCrumbs>(() => {
```
Comment (pleek91) [inline src/components/ArtifactCard.vue:77]: So with flow runs when we show crumbs we omit the flow if we're on the flow page. But here you're doing the opposite. 

I'm probably missing some context. But won't all the artifacts have the same crumbs?

**Comment by pleek91** [comment]
I might stack these dates on the condensed view so that things line up a little better. 
image

File: src/utilities/dates.ts (hunk 1 above)
Comment (znicholasbrown) [reply src/utilities/dates.ts:9]: In your defense the example isn't great 😆

File: src/components/ArtifactCard.vue (hunk 2 above)
Comment (znicholasbrown) [reply src/components/ArtifactCard.vue]: You know I can't find anything about that but I'm also not confident this is semantically helpful - I've dumped both

File: src/components/ArtifactCard.vue (hunk 3 above)
Comment (znicholasbrown) [reply src/components/ArtifactCard.vue:20]: I actually think we should be doing this more but I've removed this instance in favor of something holistic down the line. I don't think this is an issue for screen readers in general (I might be wrong here) but the idea is to provide a locale-compliant browser-supplied timestamp in addition to whatever our display format is. This is particularly helpful in places where we use a relative datetime format (this used to which is what spurred the idea)

Comment (pleek91) [reply src/components/ArtifactCard.vue:20]: Ah yes for relative times that would totally make sense 👍

File: src/components/ArtifactCard.vue (hunk 4 above)
Comment (znicholasbrown) [reply src/components/ArtifactCard.vue:78]: Ah nice suggestion 👍🏻

File: src/components/ArtifactCard.vue (hunk 6 above)
Comment (znicholasbrown) [reply src/components/ArtifactCard.vue:71]: Ah maybe this isn't as readable as it should be but the idea is exactly the same: don't show crumbs for pages we're on. Starting with no crumbs, we check if the flow run exists and that its ID isn't in the current route:
```ts
    if (flowRun.value && !routeIncludes.value.flowRun) {
      internalCrumbs.push({
        text: flowRun.value.name ?? '',
        to: routes.flowRun(flowRun.value.id),
      })
    }
```

Comment (znicholasbrown) [reply src/components/ArtifactCard.vue:71]: On the flow run page, for instance, all the flow run crumbs would be the same (and so they would be omitted) but the task run crumbs would be different

Comment (pleek91) [reply src/components/ArtifactCard.vue:71]: Ah okay that makes sense. I was just reading this wrong. 

I might break up the `routeIncludes` computed into two checks that are similar to `const viewingFlowRun = computed(() => ...)`. Might make it a little more readable

Comment (znicholasbrown) [reply src/components/ArtifactCard.vue:74]: Makes sense - modified that

**Review by pleek91 — APPROVED** [review]

**Pull request merged**